### PR TITLE
Unit tests for feat partial mb execution

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Added unit tests for feat/partial-mb-execution

### CODE REVIEW

File: process/block/baseProcess_test.go [hunk 1]
```diff
@@ -789,6 +789,157 @@ func TestVerifyStateRoot_ShouldWork(t *testing.T) {
...
+	err = bp.SetProcessingTypeAndConstructionStateForScheduledMb(miniBlockHeader, processedMiniBlocksDestMeInfo)
+	assert.Nil(t, err)
+	assert.Equal(t, int32(block.PartialExecuted), miniBlockHeader.GetConstructionState())
+	assert.Equal(t, int32(block.Scheduled), miniBlockHeader.GetProcessingType())
+}
+
+func Test_setProcessingTypeAndConstructionStateForNormalMb(t *testing.T) {
+	t.Parallel()
+
+	arguments := CreateMockArguments(createComponentHolderMocks())
```
Comment (sstanculeanu) [inline process/block/baseProcess_test.go:883]: suggestion: create `arguments` inside each `t.Run` and add `t.Paralel()` on both of them

File: process/block/processedMb/processedMiniBlocks_test.go [hunk 2]
```diff
@@ -81,3 +81,26 @@ func TestProcessedMiniBlocks_ConvertSliceToProcessedMiniBlocksMap(t *testing.T)
...
+
+	processedMiniBlockInfo, processedMetaHash := pmb.GetProcessedMiniBlockInfo(nil)
+	assert.Nil(t, processedMetaHash)
+	assert.False(t, processedMiniBlockInfo.IsFullyProcessed)
+	assert.Equal(t, int32(-1), processedMiniBlockInfo.IndexOfLastTxProcessed)
+
+	processedMiniBlockInfo, processedMetaHash = pmb.GetProcessedMiniBlockInfo(mbHash)
+	assert.Equal(t, metaHash, processedMetaHash)
+	assert.True(t, processedMiniBlockInfo.IsFullyProcessed)
+	assert.Equal(t, int32(69), processedMiniBlockInfo.IndexOfLastTxProcessed)
```
Comment (sstanculeanu) [inline process/block/processedMb/processedMiniBlocks_test.go:105]:
```suggestion
	assert.Equal(t, processedMbInfo.IndexOfLastTxProcessed, processedMiniBlockInfo.IndexOfLastTxProcessed)
```

File: process/block/baseProcess_test.go [hunk 3]
```diff
@@ -789,6 +789,157 @@ func TestVerifyStateRoot_ShouldWork(t *testing.T) {
...
+
+	t.Run("execute all scheduled txs fail", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/baseProcess_test.go:885]: Changed the message

File: process/block/baseProcess_test.go [hunk 4]
```diff
@@ -789,6 +789,157 @@ func TestVerifyStateRoot_ShouldWork(t *testing.T) {
...
+
+		processedMiniBlocksDestMeInfo[string(mbHash)] = processedMbInfo
+
+		err = bp.SetProcessingTypeAndConstructionStateForNormalMb(miniBlockHeader, processedMiniBlocksDestMeInfo)
+		assert.Nil(t, err)
+		assert.Equal(t, int32(block.PartialExecuted), miniBlockHeader.GetConstructionState())
+		assert.Equal(t, int32(block.Normal), miniBlockHeader.GetProcessingType())
+	})
+
+	t.Run("execute all scheduled txs fail", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/baseProcess_test.go:911]: Changed the message

File: process/block/displayBlock_test.go [hunk 5]
```diff
@@ -89,3 +89,19 @@ func TestDisplayBlock_DisplayTxBlockBody(t *testing.T) {
...
 }
+
+func TestDisplayBlock_GetConstructionStateAsString(t *testing.T) {
+	miniBlockHeader := &block.MiniBlockHeader{}
+
+	_ = miniBlockHeader.SetConstructionState(int32(block.Proposed))
+	str := getConstructionStateAsString(miniBlockHeader)
+	assert.Equal(t, "Proposed_", str)
+
+	_ = miniBlockHeader.SetConstructionState(int32((block.PartialExecuted)))
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/displayBlock_test.go:100]: Done

File: process/block/displayBlock_test.go [hunk 6]
```diff
@@ -89,3 +89,19 @@ func TestDisplayBlock_DisplayTxBlockBody(t *testing.T) {
...
+	str = getConstructionStateAsString(miniBlockHeader)
+	assert.Equal(t, "Partial_", str)
+
+	_ = miniBlockHeader.SetConstructionState(int32((block.Final)))
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/displayBlock_test.go:104]: Done

File: process/block/baseProcess_test.go (hunk 1 above)
Comment (SebastianMarian) [reply process/block/baseProcess_test.go:883]: Done

File: process/block/processedMb/processedMiniBlocks_test.go [hunk 7]
```diff
@@ -81,3 +81,26 @@ func TestProcessedMiniBlocks_ConvertSliceToProcessedMiniBlocksMap(t *testing.T)
...
+	pmb.SetProcessedMiniBlockInfo(metaHash, mbHash, processedMbInfo)
+
+	processedMiniBlockInfo, processedMetaHash := pmb.GetProcessedMiniBlockInfo(nil)
+	assert.Nil(t, processedMetaHash)
+	assert.False(t, processedMiniBlockInfo.IsFullyProcessed)
+	assert.Equal(t, int32(-1), processedMiniBlockInfo.IndexOfLastTxProcessed)
+
+	processedMiniBlockInfo, processedMetaHash = pmb.GetProcessedMiniBlockInfo(mbHash)
+	assert.Equal(t, metaHash, processedMetaHash)
+	assert.True(t, processedMiniBlockInfo.IsFullyProcessed)
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/processedMb/processedMiniBlocks_test.go:104]: Done

**Review by sstanculeanu — APPROVED** [review]

File: process/block/baseProcess_test.go [hunk 8]
```diff
@@ -789,6 +789,157 @@ func TestVerifyStateRoot_ShouldWork(t *testing.T) {
...
+
+	miniBlockHeader.SenderShardID = 1
+	err = bp.SetProcessingTypeAndConstructionStateForScheduledMb(miniBlockHeader, processedMiniBlocksDestMeInfo)
+	assert.Nil(t, err)
+	assert.Equal(t, int32(block.Final), miniBlockHeader.GetConstructionState())
+	assert.Equal(t, int32(block.Scheduled), miniBlockHeader.GetProcessingType())
+
+	processedMiniBlocksDestMeInfo[string(mbHash)] = processedMbInfo
+
+	miniBlockHeader.SenderShardID = 1
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/baseProcess_test.go:873]: Done

File: process/block/shardblock_test.go [hunk 9]
```diff
@@ -5044,3 +5044,46 @@ func TestShardProcessor_createMiniBlocks(t *testing.T) {
 	require.Nil(t, err)
 	require.True(t, called.IsSet())
 }
+
+func TestShardProcessor_RollBackProcessedMiniBlockInfo(t *testing.T) {
+	t.Parallel()
+
+	coreComponents, dataComponents, bootstrapComponents, statusComponents := createComponentHolderMocks()
+	arguments := CreateMockArguments(coreComponents, dataComponents, bootstrapComponents, statusComponents)
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/block/shardblock_test.go:5052]: Done

**Review by ssd04 — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Pull request merged**